pkg/frontend/adminactions/resources_list.go

```diff
@@ -0,0 +1,77 @@
+package adminactions
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	mgmtcompute "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-03-01/compute"
+	"github.com/Azure/go-autorest/autorest/azure"
+
+	"github.com/Azure/ARO-RP/pkg/util/arm"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/compute"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/features"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/network"
+	"github.com/Azure/ARO-RP/pkg/util/refreshable"
+)
+
+func (a *adminactions) ResourcesList(ctx context.Context, fpAuth refreshable.Authorizer,
+	subscriptionID, clusterRG, vNetID string) ([]byte, error) {
+
+	resourcesClient := features.NewResourcesClient(subscriptionID, fpAuth)
+	vmClient := compute.NewVirtualMachinesClient(subscriptionID, fpAuth)
+	vnetClient := network.NewVirtualNetworksClient(subscriptionID, fpAuth)
```

Comment:
I think all these clients should go into the adminactions struct and New() should create them, there is not a big overhead and it will make it a lot easy to unit test adminactions

Reply:
That was in the previous refactoring PR, but I thought we didn't want it because of overhead (make them all every time for every action). But if that's not a big deal, then I agree it would be cleaner to make them in one place.

Reply:
@asalkeld Done